Sources/Sentry/SentryProfiler.mm

```diff
@@ -256,36 +421,38 @@ - (SentryEnvelopeItem *)buildEnvelopeItemForTransaction:(SentryTransaction *)tra
     profile[@"device_is_emulator"] = @(isSimulatorBuild());
     profile[@"device_physical_memory_bytes"] =
         [@(NSProcessInfo.processInfo.physicalMemory) stringValue];
-    profile[@"environment"] = hub.scope.environmentString ?: hub.getClient.options.environment ?: kSentryDefaultEnvironment;
-    profile[@"platform"] = transaction.platform;
-    profile[@"transaction_id"] = transaction.eventId.sentryIdString;
-    profile[@"trace_id"] = transaction.trace.context.traceId.sentryIdString;
-    profile[@"profile_id"] = [[SentryId alloc] init].sentryIdString;
-    profile[@"transaction_name"] = transaction.transaction;
-    profile[@"duration_ns"] = [@(getDurationNs(_startTimestamp, getAbsoluteTime())) stringValue];
+    const auto profileID = [[SentryId alloc] init];
+    profile[@"profile_id"] = profileID.sentryIdString;
+    const auto profileDuration = getDurationNs(_startTimestamp, _endTimestamp);
+    profile[@"duration_ns"] = [@(profileDuration) stringValue];
+    profile[@"truncation_reason"] = profilerTruncationReasonName(_truncationReason);
 
     const auto bundle = NSBundle.mainBundle;
     profile[@"version_code"] = [bundle objectForInfoDictionaryKey:(NSString *)kCFBundleVersionKey];
     profile[@"version_name"] = [bundle objectForInfoDictionaryKey:@"CFBundleShortVersionString"];
 
 #    if SENTRY_HAS_UIKIT
     auto relativeFrameTimestampsNs = [NSMutableArray array];
-    [frameInfo.frameTimestamps enumerateObjectsUsingBlock:^(
+    [_frameInfo.frameTimestamps enumerateObjectsUsingBlock:^(
         NSDictionary<NSString *, NSNumber *> *_Nonnull obj, NSUInteger idx, BOOL *_Nonnull stop) {
         const auto begin = (uint64_t)(obj[@"start_timestamp"].doubleValue * 1e9);
         if (begin < _startTimestamp) {
             return;
         }
         const auto end = (uint64_t)(obj[@"end_timestamp"].doubleValue * 1e9);
+        const auto relativeEnd = getDurationNs(_startTimestamp, end);
+        if (relativeEnd > profileDuration) {
+            return;
```

Comment:
If this happens doesn't mean we should set `*stop = YES` to prevent further enumeration since any subsequent timestamps would also be beyond the duration? Or are the timestamps not ordered that way?

Reply:
I think the timestamps are ordered that way, yes. Which means this _should_ only happen once, I don't think there can be overlapping frame renders. Is that true? If so I can at least add a comment here talking about that.

Reply:
I think adding a comment would make sense. Maybe even adding a log message for this edge case.